app/src/main/java/com/github/se/cyrcle/ui/review/AllReviewsScreen.kt

```diff
@@ -94,24 +101,25 @@ fun AllReviewsScreen(
                             Modifier.fillMaxWidth()
                                 .padding(horizontal = 16.dp)
                                 .testTag("ReviewList")) {
-                          reviewViewModel.getReviewsByParking(selectedParking.uid)
-                          items(reviewViewModel.parkingReviews.value.size) { index ->
-                            val curReview = reviewViewModel.parkingReviews.value[index]
-                            userViewModel.getUserById(curReview.owner)
-                            val uidOfOwner =
-                                userViewModel.currentUser.value?.public?.username ?: "none"
-                            if (index == selectedCardIndex) {
+                          items(items = reviewsList) { curReview ->
+                            val index = reviewsList.indexOf(curReview)
+                            val defaultUsername = stringResource(R.string.undefined_username)
+                            val uidOfOwner = remember { mutableStateOf(defaultUsername) }
+                            userViewModel.getUserById(curReview.owner) {
+                              uidOfOwner.value = it.public.username
```

Comment:
just to verify w.r.t edge cases, is it possible that the object here may be null? (since error handling for null case was handled in the original code)

Reply:
yes, if the viewmodel getUserById fails because of a null arg, or because the owner does not exist anymore, it will call the onFailure, which is not defined, so nothing happens

Reply:
Figured as much, good to see this was thought of and handled correctly